From 7716f54098f00221ac3e7218ddd6d5efa32c72e1 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:19 -0400
Subject: [tools] perf/list: Show error if tracepoints not available

Message-id: <1408462094-14194-211-git-send-email-jolsa@redhat.com>
Patchwork-id: 88209
O-Subject: [PATCH RHEL7.1 BZ1131394 210/325] perf list: Show error if tracepoints not available
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit f11cfc6f294dbd83b0d58037404df2bd16066238
Author: Pekka Enberg <penberg@kernel.org>
Date: Tue Oct 15 23:07:27 2013 +0300

description
===========
Tracepoints are not visible in "perf list" on Fedora 19 because regular
users have no permission to /sys/kernel/debug by default. Show an error
message so that the user knows about it instead of assuming that
tracepoints are not supported on the system.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/parse-events.c b/tools/perf/util/parse-events.c
index 9812531..c90e55c 100644
--- a/tools/perf/util/parse-events.c
+++ b/tools/perf/util/parse-events.c
@@ -998,8 +998,10 @@ void print_tracepoint_events(const char *subsys_glob, const char *event_glob,
  char evt_path[MAXPATHLEN];
  char dir_path[MAXPATHLEN];
 
- if (debugfs_valid_mountpoint(tracing_events_path))
+ if (debugfs_valid_mountpoint(tracing_events_path)) {
+  printf("  [ Tracepoints not available: %s ]\n", strerror(errno));
   return;
+ }
 
  sys_dir = opendir(tracing_events_path);
  if (!sys_dir)
-- 
1.7.1